core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -151,20 +151,21 @@ public InternalEngine(EngineConfig engineConfig, boolean skipInitialTranslogReco
                     throw e;
                 }
             }
+
             this.translog = translog;
             manager = createSearcherManager();
             this.searcherManager = manager;
             this.versionMap.setManager(searcherManager);
             try {
-                if (skipInitialTranslogRecovery) {
+                if (openMode == EngineConfig.OpenMode.OPEN_INDEX_CREATE_TRANSLOG) {
                     // make sure we point at the latest translog from now on..
                     commitIndexWriter(writer, translog, lastCommittedSegmentInfos.getUserData().get(SYNC_COMMIT_ID));
-                } else {
-                    recoverFromTranslog(engineConfig, translogGeneration);
                 }
             } catch (IOException | EngineException ex) {
                 throw new EngineCreationFailureException(shardId, "failed to recover from translog", ex);
```

Comment:
this is exception message is misleading , but I think the block can go away, see above comment.